buildSrc/src/main/kotlin/kotest-native-conventions.gradle.kts

```diff
@@ -20,74 +21,31 @@ kotlin {
-
-         val desktopMain by creating {
-            dependsOn(commonMain)
+      // FIXME: The "desktop" intermediate source set can be integrated into "native". In this case
+      //     the following block can be replaced with `applyDefaultHierarchyTemplate()`.
+      applyHierarchyTemplate(KotlinHierarchyTemplate.default) {
+         group("common") {
+            group("desktop") {
+               withNative()
+            }
          }
+      }
 
-         val macosX64Main by getting { dependsOn(desktopMain) }
-         val macosArm64Main by getting { dependsOn(desktopMain) }
-
-         val mingwX64Main by getting { dependsOn(desktopMain) }
-
-         val linuxX64Main by getting { dependsOn(desktopMain) }
-         val linuxArm64Main by getting { dependsOn(desktopMain) }
-
-         val iosX64Main by getting { dependsOn(desktopMain) }
-         val iosArm64Main by getting { dependsOn(desktopMain) }
-         val iosArm32Main by getting { dependsOn(desktopMain) }
-         val iosSimulatorArm64Main by getting { dependsOn(desktopMain) }
-
-         val watchosArm32Main by getting { dependsOn(desktopMain) }
-         val watchosArm64Main by getting { dependsOn(desktopMain) }
-         val watchosX86Main by getting { dependsOn(desktopMain) }
-         val watchosX64Main by getting { dependsOn(desktopMain) }
-         val watchosSimulatorArm64Main by getting { dependsOn(desktopMain) }
-
-         val tvosX64Main by getting { dependsOn(desktopMain) }
-         val tvosArm64Main by getting { dependsOn(desktopMain) }
-         val tvosSimulatorArm64Main by getting { dependsOn(desktopMain) }
-
-         // Test sourcesets
-         val commonTest by getting
-
-         val nativeTest by creating { dependsOn(commonTest) }
-
-         val macosX64Test by getting { dependsOn(nativeTest) }
-         val macosArm64Test by getting { dependsOn(nativeTest) }
-
-         val mingwX64Test by getting { dependsOn(nativeTest) }
-
-         val linuxX64Test by getting { dependsOn(nativeTest) }
-         val linuxArm64Test by getting { dependsOn(nativeTest) }
-
-         val iosX64Test by getting { dependsOn(nativeTest) }
-         val iosArm64Test by getting { dependsOn(nativeTest) }
-         val iosArm32Test by getting { dependsOn(nativeTest) }
-         val iosSimulatorArm64Test by getting { dependsOn(nativeTest) }
-
-         val watchosArm32Test by getting { dependsOn(nativeTest) }
-         val watchosArm64Test by getting { dependsOn(nativeTest) }
-         val watchosX86Test by getting { dependsOn(nativeTest) }
-         val watchosX64Test by getting { dependsOn(nativeTest) }
-         val watchosSimulatorArm64Test by getting { dependsOn(nativeTest) }
-
-         val tvosX64Test by getting { dependsOn(nativeTest) }
-         val tvosArm64Test by getting { dependsOn(nativeTest) }
-         val tvosSimulatorArm64Test by getting { dependsOn(nativeTest) }
+      // FIXME: WORKAROUND https://github.com/Kotlin/kotlinx.coroutines/issues/3968
```

Comment:
@OliverO2 FYI coroutines 1.8.0 is out now with a fix for this bug, so the workaround is no longer necessary

Reply:
Yep. Thanks for the reminder. I'll get back to this as soon as someone is ready for a complete review (but no sooner than March, 4th due to other commitments). Looking forward to updating this and possibly other dependencies then.